### DISCUSSION TITLE
[Question] Function window_center

### DISCUSSION THREAD

**Discussion by ingdesarr1** [flet-dev/flet#2807] [Q&A] [answered]
### Question

Good morning. I'm trying to use the function window_center(), using the event on_window_event to detect when the window move to other position, but not working. I even use the function window_destroy, to probe if move the window works, but not working too. How can I do that? Thanks

### Code sample

```python
import flet as ft

def main(page:ft.Page):

    def mover(e):
        print(e.data)
        if e.data == 'moved':
            # print("hola")
            page.window_destroy()
            # page.update()

    page.window_focused = True
    page.window_resizable = False
    page.window_movable = False
    # page.window_maximized = True
    page.window_maximizable = False
    page.window_center()
    page.theme_mode = "LIGHT"
    page.on_window_event = mover

ft.app(target=main, assets_dir="assets")
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by burhansvural** [comment]
@ingdesarr1 

very easy :

```python
def main(page: ft.Page):
    x=page.window_top
    y=page.window_left

    def sss(e):
        nonlocal x,y
        if e.control.window_top!=x or e.control.window_left!=y:
            print("Page moved")
            x=e.control.window_top
            y=e.control.window_left

    page.on_window_event=sss
```

**Reply by ingdesarr1** [reply]
Don't work. The window is not center.

**Reply by burhansvural** [reply]
In the above example, how the page moved process works is simply illustrated. If you want to center the window on the screen, you should use page.window_center(). If you are concerned about monitor resolution, you should use the "screeninfo" library.

**Reply by ingdesarr1** [reply]
Yes, I use the function, but don't work, and I don't know why.

**Reply by burhansvural** [reply]
```python
import flet as ft

def main(page:ft.Page):

    x=page.window_top
    y=page.window_left

    def sss(e):
        nonlocal x,y
        print(e.data)
        if e.control.window_top!=x or e.control.window_left!=y:            
            e.data="Moved"
            x=e.control.window_top
            y=e.control.window_left
            print(e.data)
    
    page.theme_mode = ft.ThemeMode.LIGHT    
    page.window_width=800
    page.window_height=600
    page.on_window_event = sss
    page.add(ft.Text("Hello"))
    page.window_center()  
    

ft.app(target=main, assets_dir="assets")
```

**Reply by ingdesarr1** [reply, marked as answer]
I found the error. there is a problem with this function:

page.window_movable = False

With this property, the page functions don't work, because I commented this line and all the functions work
